bindings/src/main/java/io/pravega/storage/filesystem/FileSystemChunkStorage.java

```diff
@@ -0,0 +1,317 @@
+
+    //endregion
+
+    //region capabilities
+
+    @Override
+    public boolean supportsConcat() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsAppend() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsTruncation() {
+        return false;
+    }
+
+    //endregion
+
+    //region
+
+    @VisibleForTesting
+    protected FileChannel getFileChannel(Path path, StandardOpenOption openOption) throws IOException {
+        try {
+            return FileChannel.open(path, openOption);
+        } catch (NoSuchFileException e) {
+            throw new FileNotFoundException(path.toString());
+        }
+    }
+
+    @VisibleForTesting
+    protected long getFileSize(Path path) throws IOException {
+        try {
+            return Files.size(path);
+        } catch (NoSuchFileException e) {
+            throw new FileNotFoundException(path.toString());
+        }
+    }
+
+    @Override
+    protected ChunkInfo doGetInfo(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        try {
+            PosixFileAttributes attrs = Files.readAttributes(Paths.get(config.getRoot(), chunkName),
+                    PosixFileAttributes.class);
+            ChunkInfo information = ChunkInfo.builder()
+                    .name(chunkName)
+                    .length(attrs.size())
+                    .build();
+
+            return information;
+        } catch (IOException e) {
+            throwExeption(chunkName, "doGetInfo", e);
+        }
+        throw new ChunkStorageException(chunkName, "Unreachable code");
+    }
+
+    @Override
+    protected ChunkHandle doCreate(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        try {
+            FileAttribute<Set<PosixFilePermission>> fileAttributes = PosixFilePermissions.asFileAttribute(FileSystemUtils.READ_WRITE_PERMISSION);
+
+            Path path = Paths.get(config.getRoot(), chunkName);
+            Path parent = path.getParent();
+            assert parent != null;
+            Files.createDirectories(parent);
+            Files.createFile(path, fileAttributes);
+
+        } catch (IOException e) {
+            throwExeption(chunkName, "doCreate", e);
+        }
+
+        return ChunkHandle.writeHandle(chunkName);
+    }
+
+    public void throwExeption(String chunkName, String message, Exception e) throws ChunkStorageException {
+        if (e instanceof FileNotFoundException || e instanceof NoSuchFileException) {
+            throw new ChunkNotFoundException(chunkName, message);
+        }
+        if (e instanceof FileAlreadyExistsException) {
+            throw new ChunkAlreadyExistsException(chunkName, message);
+        }
+        throw new ChunkStorageException(chunkName, message, e);
+    }
+
+    @Override
+    protected boolean checkExists(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        return Files.exists(Paths.get(config.getRoot(), chunkName));
```

Comment:
This doesn't really throw `ChunkStorageException`, does it?

Reply:
It doesn't have to

Reply:
fixed